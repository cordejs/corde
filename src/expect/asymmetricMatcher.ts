import { typeOf } from "../utils/typeOf";

export class AsymmetricMatcher {
  private _classType: any[];

  constructor(...classType: any[]) {
    this._classType = classType;
  }

  isSpecified() {
    return this._classType.length > 0;
  }

  toString() {
    if (this.isSpecified()) {
      const names = this._classType.map((type) => this.getTypeName(type));
      return `any(${names.join(", ")})`;
    }
    return "any";
  }

  getTypeName(classType: any) {
    if (classType === Number) {
      return "Number";
    }

    if (classType === Date) {
      return "Date";
    }

    if (classType === BigInt) {
      return "Bigint";
    }

    if (classType === String) {
      return "String";
    }

    if (classType === Boolean) {
      return "Boolean";
    }

    if (classType === Array) {
      return "Array";
    }

    if (classType === Object) {
      return "Object";
    }

    if (classType === Function) {
      return "Function";
    }

    if (classType === Symbol) {
      return "Symbol";
    }

    if (typeof classType === "function") {
      return classType.name;
    }

    return classType;
  }

  getTypes() {
    return this._classType;
  }

  matchType(...classType: any[]) {
    return (
      this._classType === undefined ||
      this._classType === null ||
      this._classType.length === 0 ||
      classType.length === 0 ||
      this._classType.some((type) => classType.includes(type))
    );
  }

  matchValue(value: any) {
    return (
      this.isSpecified() ||
      this._classType.every((classType) => this._isValueOfType(classType, value))
    );
  }

  private _isValueOfType(classType: any, value: any) {
    if (classType === Number) {
      return typeOf(value) === "number" || value instanceof Number;
    }

    if (classType === BigInt) {
      return typeOf(value) === "bigint" || value instanceof BigInt;
    }

    if (classType === String) {
      return typeOf(value) === "string" || value instanceof String;
    }

    if (classType === Boolean) {
      return typeOf(value) === "boolean" || value instanceof Boolean;
    }

    if (classType === Array) {
      return typeOf(value) === "array";
    }

    if (classType === Object) {
      return typeOf(value) === "object";
    }

    if (classType === Function) {
      return typeOf(value) === "function" || value instanceof Function;
    }

    if (classType === Symbol) {
      return typeOf(value) === "symbol";
    }

    if (typeOf(value) === "object") {
      return value instanceof classType;
    }

    return false;
  }
}

export function any(...classType: any[]) {
  return new AsymmetricMatcher(...classType);
}
