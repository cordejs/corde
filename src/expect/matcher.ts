import { ColorResolvable } from "discord.js";
import { testCollector } from "../common/testCollector";
import {
  IEmoji,
  IMessageIdentifier,
  IMessageEditedIdentifier,
  IMessageEmbed,
  Primitive,
  IRoleIdentifier,
  ITestReport,
  ICordeBot,
  TestFunctionType,
  IMessageMatches,
  IRoleMatches,
} from "../types";
import { Colors } from "../utils/colors";
import { RolePermission } from "../utils/permission";
import {
  ToAddReaction,
  ToDeleteRole,
  ToRenameRole,
  ToReturn,
  ToSetRoleColor,
  ToSetRoleHoist,
  ToSetRoleMentionable,
  ToSetRolePosition,
  ToEditMessage,
  ToPinMessage,
  ToRemoveReaction,
  ToUnPinMessage,
  ToSetRolePermission,
  ToEmbedMatch,
  ToMessageContentContains,
} from "./matches";
import { ExpectTest } from "./matches/expectTest";
import { buildReportMessage, resolveName, stringIsNullOrEmpty } from "../utils";
import { getStackTrace } from "../utils/getStackTrace";
import { runtime } from "../common/runtime";
import { IMacherContructorArgs, MayReturnMatch, IExpectTestBaseParams } from "../types";
import { IToHaveResult } from "./matches/toHaveResult";

class BaseMatcher {
  protected _commandName: unknown;
  protected _isNot: boolean;
  protected _isCascade: boolean;
  protected _guildId?: string;
  protected _channelId?: string;
  protected _channelIdToSendCommand?: string;

  constructor({
    commandName,
    isNot,
    isCascade,
    channelId,
    guildId,
    channelIdToSendCommand,
  }: IMacherContructorArgs) {
    this._commandName = commandName;
    this._isNot = isNot ?? false;
    this._isCascade = isCascade ?? false;
    this._guildId = guildId;
    this._channelId = channelId;
    this._channelIdToSendCommand = channelIdToSendCommand;
  }

  // Trace can not me added inside operationFactory because it do,
  // it will get irrelevant data.

  protected async operationFactory<T extends ExpectTest>(
    trace: string,
    type: new (params: IExpectTestBaseParams) => T,
    cordeBot: ICordeBot,
    ...params: Parameters<T["action"]>
  ): Promise<ITestReport> {
    const commandName = await resolveName(this._commandName);

    const op = new type({
      cordeBot,
      command: commandName,
      isNot: this._isNot,
      timeout: runtime.timeOut,
      isCascade: this._isCascade,
      guildId: this._guildId ?? runtime.guildId,
      channelId: this._channelId ?? runtime.channelId,
      channelIdToSendCommand: this._channelIdToSendCommand,
    });

    if (
      !this._isCascade &&
      (commandName == undefined ||
        (typeof commandName === "string" && stringIsNullOrEmpty(commandName)))
    ) {
      return {
        pass: false,
        message: buildReportMessage("command can not be null or an empty string"),
        testName: op.toString(),
      };
    }

    const report = await op.action(...params);

    if (!report) {
      return {
        pass: false,
        message: buildReportMessage("no report was provided by the test"),
        testName: op.toString(),
        trace,
      };
    }

    if (report.pass) {
      return report;
    }

    report.trace = trace;
    return report;
  }

  protected returnOrAddToCollector(testFunction: TestFunctionType): any {
    if (this._isCascade) {
      return testFunction;
    }

    return testCollector.addTestFunction(testFunction);
  }
}

export class MessageMatches<TReturn extends MayReturnMatch>
  extends BaseMatcher
  implements IMessageMatches<TReturn>
{
  toEmbedMatch(embed: IMessageEmbed): TReturn {
    const trace = getStackTrace(undefined, true, "toEmbedMatch");
    return this.returnOrAddToCollector((cordeBot) =>
      this.operationFactory(trace, ToEmbedMatch, cordeBot, embed),
    );
  }

  toMessageContentContains(expectedContent: string): TReturn {
    const trace = getStackTrace(undefined, true, "toMessageContentContains");
    return this.returnOrAddToCollector((cordeBot) =>
      this.operationFactory(trace, ToMessageContentContains, cordeBot, expectedContent),
    );
  }

  toReturn(expect: Primitive | IMessageEmbed) {
    const trace = getStackTrace(undefined, true, "toReturn");
    return this.returnOrAddToCollector((cordeBot) =>
      this.operationFactory(trace, ToReturn, cordeBot, expect),
    );
  }

  toEditMessage(
    newValue: Primitive | IMessageEmbed,
    messageIdentifier?: IMessageEditedIdentifier | string,
  ) {
    const trace = getStackTrace(undefined, true, "toEditMessage");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToEditMessage, cordeBot, newValue, messageIdentifier);
    });
  }

  toPin(messageIdentifier: string | IMessageIdentifier) {
    const trace = getStackTrace(undefined, true, "toPin");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToPinMessage, cordeBot, messageIdentifier);
    });
  }

  toUnPin(messageIdentifier: string | IMessageIdentifier) {
    const trace = getStackTrace(undefined, true, "toUnPin");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToUnPinMessage, cordeBot, messageIdentifier);
    });
  }

  toAddReaction(
    emojis: string[] | IEmoji[] | (string | IEmoji)[],
    messageIdentifier?: string | IMessageIdentifier,
  ) {
    const trace = getStackTrace(undefined, true, "toAddReaction");
    return this.returnOrAddToCollector((cordeBot) =>
      this.operationFactory(trace, ToAddReaction, cordeBot, emojis, messageIdentifier),
    );
  }

  toRemoveReaction(
    emojis: string[] | IEmoji[] | (string | IEmoji)[],
    messageIdentifier?: string | IMessageIdentifier,
  ) {
    const trace = getStackTrace(undefined, true, "toRemoveReaction");
    return this.returnOrAddToCollector((cordeBot) =>
      this.operationFactory(trace, ToRemoveReaction, cordeBot, emojis, messageIdentifier),
    );
  }
}

export class ToHaveResultMatcher extends BaseMatcher {
  toHaveResult(...tests: TestFunctionType[]): void {
    const trace = getStackTrace(undefined, true, "todoInCascade");
    testCollector.addTestFunction((cordeBot) => {
      // Encapsulate the functions inside another so it do not be executed.
      const testEnhanced = tests.map((test) => () => test(cordeBot));
      return this.operationFactory(trace, IToHaveResult, cordeBot, ...testEnhanced);
    });
  }
}

export class RoleMatchesImpl<TReturn extends MayReturnMatch>
  extends BaseMatcher
  implements IRoleMatches<TReturn>
{
  toRenameRole(newName: string, roleIdentifier: IRoleIdentifier | string) {
    const trace = getStackTrace(undefined, true, "toRenameRole");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToRenameRole, cordeBot, newName, roleIdentifier);
    });
  }

  toSetRoleColor(color: ColorResolvable | Colors, roleIdentifier: string | IRoleIdentifier) {
    const trace = getStackTrace(undefined, true, "toSetRoleColor");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToSetRoleColor, cordeBot, color, roleIdentifier);
    });
  }

  toDeleteRole(roleIdentifier: string | IRoleIdentifier) {
    const trace = getStackTrace(undefined, true, "toDeleteRole");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToDeleteRole, cordeBot, roleIdentifier);
    });
  }

  toSetRoleMentionable(mentionable: boolean, roleIdentifier: string | IRoleIdentifier) {
    const trace = getStackTrace(undefined, true, "toSetRoleMentionable");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(
        trace,
        ToSetRoleMentionable,
        cordeBot,
        mentionable,
        roleIdentifier,
      );
    });
  }

  toSetRoleHoist(hoist: boolean, roleIdentifier: string | IRoleIdentifier) {
    const trace = getStackTrace(undefined, true, "toSetRoleHoist");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToSetRoleHoist, cordeBot, hoist, roleIdentifier);
    });
  }

  toSetRolePosition(newPosition: number, roleIdentifier: string | IRoleIdentifier) {
    const trace = getStackTrace(undefined, true, "toSetRolePosition");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(trace, ToSetRolePosition, cordeBot, newPosition, roleIdentifier);
    });
  }

  toSetRolePermission(roleIdentifier: string | IRoleIdentifier, ...permissions: RolePermission[]) {
    const trace = getStackTrace(undefined, true, "toSetRolePermission");
    return this.returnOrAddToCollector((cordeBot) => {
      return this.operationFactory(
        trace,
        ToSetRolePermission,
        cordeBot,
        roleIdentifier,
        permissions,
      );
    });
  }
}
