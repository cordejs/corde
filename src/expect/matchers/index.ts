export * from "./toBeArray";
export * from "./toBeBigInt";
export * from "./toBeBoolean";
export * from "./toBeDate";
export * from "./toBeDefined";
export * from "./toBeEmptyArray";
export * from "./toBeEmptyString";
export * from "./toBeFunction";
export * from "./toBeGreaterOrEqualThan";
export * from "./toBeGreaterThan";
export * from "./toBeLessOrEqualThan";
export * from "./toBeLessThan";
export * from "./toBeNaN";
export * from "./toBeNothing";
export * from "./toBeNull";
export * from "./toBeUndefined";
export * from "./toBeNumber";
export * from "./toBePrimitive";
export * from "./toBeString";
export * from "./toBeSymbol";
export * from "./toBeValidDate";
export * from "./toLength";
export * from "./toBeInstanceOf";
export * from "./toBeNegativeInfinity";
export * from "./toBePositiveInfinity";
export * from "./toBe";
