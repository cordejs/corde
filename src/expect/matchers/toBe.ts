import chalk from "chalk";
import { diff } from "jest-diff";
import { ITestProps } from "../../types";
import { typeOf } from "../../utils/typeOf";
import { matcherUtils } from "../matcherUtils";

/**
 * @internal
 */
export function toBe(this: ITestProps, expected: any, received: any) {
  let pass = matcherUtils.matchValues(() => expected === received, expected, received);

  let comparator = "===";

  if (this.isNot) {
    pass = !pass;
    comparator = "!==";
  }

  const _getReportMessage = () => {
    if (typeOf(expected) === "object" && typeOf(received) === "object") {
      return diff(expected, received);
    }

    const expectedFormatted = this.formatValue(expected);
    const receivedFormatted = this.formatValue(received);

    return `expect: ${chalk.green(expectedFormatted)} ${comparator} ${chalk.red(
      receivedFormatted,
    )}`;
  };

  return {
    pass,
    message: pass
      ? ""
      : `Tip: toBe asserts identity of values. To compare only values use ${chalk.bold(
          "toEqual",
        )}\n` +
        this.createHint("received") +
        "\n" +
        `${_getReportMessage()}`,
  };
}
