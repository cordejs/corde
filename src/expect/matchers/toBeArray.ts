import chalk from "chalk";
import { ITestProps } from "../../types";
import { asymetricTypeOf, buildReportMessage } from "../../utils";
import { matcherUtils } from "../matcherUtils";

/**
 * @internal
 */
export function toBeArray(this: ITestProps, expected: any) {
  let pass = matcherUtils.match(() => Array.isArray(expected), {
    value: expected,
    validParameters: [Array],
  });
  let isNotText = "";

  if (this.isNot) {
    pass = !pass;
    isNotText = " not";
  }

  return {
    pass,
    message: pass
      ? ""
      : buildReportMessage(
          this.createHint(),
          "\n\n",
          `${this.expectedColorFn("expected")} should${isNotText} be an ${chalk.green("array")}.\n`,
          `got: ${chalk.red(asymetricTypeOf(expected))}`,
        ),
  };
}
