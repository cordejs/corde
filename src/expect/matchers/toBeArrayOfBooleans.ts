import chalk from "chalk";
import { ITestProps } from "../../types";
import { asymmetricTypeOf, buildReportMessage, every, isBoolean } from "../../utils";
import { matcherUtils } from "../matcherUtils";

/**
 * @internal
 */
export function toBeArrayOfBooleans(this: ITestProps, expected: any) {
  let pass = matcherUtils.match(
    () => {
      if (!Array.isArray(expected)) {
        return false;
      }

      return every(expected, isBoolean);
    },
    {
      value: expected,
      validParameters: [Array],
    },
  );
  let isNotText = "";

  if (this.isNot) {
    pass = !pass;
    isNotText = " not";
  }

  return {
    pass,
    message: pass
      ? ""
      : buildReportMessage(
          this.createHint(),
          "\n\n",
          `${this.expectedColorFn("expected")} should${isNotText} be an ${chalk.green(
            "array of booleans",
          )}.\n`,
          `got: ${chalk.red(asymmetricTypeOf(expected))}`,
        ),
  };
}
