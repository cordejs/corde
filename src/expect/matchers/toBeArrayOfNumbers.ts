import chalk from "chalk";
import { ITestProps } from "../../types";
import { asymetricTypeOf, buildReportMessage, every, isNumber } from "../../utils";
import { matcherUtils } from "../matcherUtils";

/**
 * @internal
 */
export function toBeArrayOfNumbers(this: ITestProps, expected: any) {
  let pass = matcherUtils.match(
    () => {
      if (!Array.isArray(expected)) {
        return false;
      }

      return every(expected, isNumber);
    },
    {
      value: expected,
      validParameters: [Array],
    },
  );
  let isNotText = "";

  if (this.isNot) {
    pass = !pass;
    isNotText = " not";
  }

  return {
    pass,
    message: pass
      ? ""
      : buildReportMessage(
          this.createHint(),
          "\n\n",
          `${this.expectedColorFn("expected")} should${isNotText} be an ${chalk.green(
            "array of numbers",
          )}.\n`,
          `got: ${chalk.red(asymetricTypeOf(expected))}`,
        ),
  };
}
