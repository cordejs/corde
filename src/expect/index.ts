/* eslint-disable no-console */
import chalk from "chalk";
import { runtime } from "../common/runtime";
import { ITestProps, ITestReport } from "../types";
import { corde } from "../types/globals";
import { getStackTrace, typeOf } from "../utils";
import { any } from "./asymmetricMatcher";
import * as matchers from "./matchers";

interface IMatcher {
  (props: ITestProps, ...args: any[]): ITestReport;
}

type KeyOfMatcher = keyof typeof matchers;

function pickFn(name: KeyOfMatcher) {
  return matchers[name] as any as IMatcher;
}

function createMatcherFn(matcher: string, isNot: boolean, expected: any, isDebug: boolean) {
  return (...args: any[]): ITestReport | void => {
    // If someone pass expect.any, we must invoke it to return
    // the Any matcher.

    if (expected === any) {
      args = [expected(), ...args];
    } else {
      args = [expected, ...args];
    }

    args = args.map((arg) => {
      if (arg === any) {
        return arg();
      }
      return arg;
    });

    const trace = getStackTrace(undefined, true, matcher);
    try {
      const matcherFn = pickFn(matcher as KeyOfMatcher);

      const props: ITestProps = {
        isNot,
        expectedColorFn: chalk.bold,
        receivedColorFn: chalk.bold,
        createHint: (...paramsName: string[]) => {
          return `expect(${chalk.green("expected")}).${isNot ? "not." : ""}${matcher}(${paramsName
            .map((param) => chalk.red(param))
            .join(",")})`;
        },
        formatValue: (value: any) => {
          if (typeof value === "symbol") {
            return value.toString();
          }

          if (typeOf(value) === "object") {
            if (Object.keys(value).length === 0) {
              return "{}";
            }

            return "{ ... }";
          }

          if (
            typeof value === "string" &&
            (value.length === 0 || (!value.startsWith("'") && !value.endsWith("'")))
          ) {
            return `'${value}'`;
          }

          if (Array.isArray(value)) {
            if (value.length > 0) {
              return "[...]";
            }
            return "[]";
          }

          return value;
        },
      };

      const report = matcherFn.bind(props, ...args)();
      if (!report.pass) {
        report.trace = trace;
      }

      runtime.internalEvents.emit("test_end", report);

      if (isDebug) {
        return report;
      }
    } catch (error) {
      const failedReport: ITestReport = {
        pass: false,
        message: handleError(error),
        trace: trace,
      };
      runtime.internalEvents.emit("test_end", failedReport);
      if (isDebug) {
        return failedReport;
      }
    }
  };
}

function handleError(error: any) {
  if (error instanceof Error) {
    return error.message;
  }
  return error;
}

function createLocalExpect(isDebug: boolean) {
  let localExpect: any = {};

  localExpect = (expected: any) => {
    const _expect: any = {};
    _expect.not = {};
    Object.getOwnPropertyNames(matchers).forEach((matcher) => {
      _expect[matcher] = createMatcherFn(matcher, false, expected, isDebug);
      _expect.not[matcher] = createMatcherFn(matcher, true, expected, isDebug);
    });
    return _expect;
  };

  localExpect.any = any;

  return localExpect;
}

export const expect = createLocalExpect(false) as corde.IExpect;

type DebugExpectType<T> = {
  [P in keyof T]: T[P] extends (...args: any[]) => any
    ? (...params: Parameters<T[P]>) => { pass: boolean; message: string }
    : DebugExpectType<T[P]>;
};

export interface IDebugExpect {
  <T extends any>(value: T): DebugExpectType<corde.IMatchersWithNot<T>>;
  any(...classType: any[]): any;
}

/**
 * Testing object for corde
 * This is not used in production
 *
 * @internal
 */
export const cordeExpect = createLocalExpect(true) as IDebugExpect;
