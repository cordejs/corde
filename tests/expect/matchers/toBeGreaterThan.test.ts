import { any } from "../../../src/expect/asymmetricMatcher";
import { toBeGreaterThan } from "../../../src/expect/matchers";
import { removeANSIColorStyle } from "../../testHelper";

describe("testing toBeGreaterThan", () => {
  it("should return true for a number be bigger than other", () => {
    expect(toBeGreaterThan({ isNot: false }, 2, 1)).toEqual({ pass: true, message: "" });
  });

  it("should return true for a number be less than other (isNot true)", () => {
    expect(toBeGreaterThan({ isNot: true }, 1, 2)).toEqual({ pass: true, message: "" });
  });

  it("should return true for a number be equal than other (isNot true)", () => {
    expect(toBeGreaterThan({ isNot: true }, 1, 1)).toEqual({ pass: true, message: "" });
  });

  it("should return true for 'expected' asymetric value", () => {
    expect(toBeGreaterThan({ isNot: false }, any(), 1)).toEqual({ pass: true, message: "" });
    expect(toBeGreaterThan({ isNot: false }, any(Number), 1)).toEqual({
      pass: true,
      message: "",
    });
  });

  it("should return true for 'received' asymetric value", () => {
    expect(toBeGreaterThan({ isNot: false }, 1, any() as any)).toEqual({
      pass: true,
      message: "",
    });
    expect(toBeGreaterThan({ isNot: false }, 1, any(Number) as any)).toEqual({
      pass: true,
      message: "",
    });
  });

  it.each([[""], ["1"], [false], [true], [[]]])(
    "should return false for 'expected' not to be a number",
    (expected) => {
      const report = toBeGreaterThan({ isNot: false }, expected, 1 as any);
      report.message = removeANSIColorStyle(report.message);
      expect(report).toMatchSnapshot();
    },
  );

  it.each([[""], ["1"], [false], [true], [[]]])(
    "should return false for 'received' not to be a number",
    (received) => {
      const report = toBeGreaterThan({ isNot: false }, 1, received as any);
      report.message = removeANSIColorStyle(report.message);
      expect(report.pass).toBeFalsy();
      expect(report).toMatchSnapshot();
    },
  );

  it("should fail due to expected be less than received", () => {
    const report = toBeGreaterThan({ isNot: false }, 1, 2);
    report.message = removeANSIColorStyle(report.message);
    expect(report.pass).toBeFalsy();
    expect(report).toMatchSnapshot();
  });

  it("should fail due to expected be equal than received", () => {
    const report = toBeGreaterThan({ isNot: false }, 1, 1);
    report.message = removeANSIColorStyle(report.message);
    expect(report.pass).toBeFalsy();
    expect(report).toMatchSnapshot();
  });

  it("should fail due to expected be bigger than received (isNot true)", () => {
    const report = toBeGreaterThan({ isNot: true }, 4, 2);
    report.message = removeANSIColorStyle(report.message);
    expect(report.pass).toBeFalsy();
    expect(report).toMatchSnapshot();
  });
});
