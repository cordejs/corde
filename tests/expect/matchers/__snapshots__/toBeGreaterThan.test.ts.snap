// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing toBeGreaterThan should fail due to expected be bigger than received (isNot true) 1`] = `
Object {
  "message": "expect([32mexpected[39m).not.toBeGreaterThan([31mreceived[39m)
result: 4 <= 2.",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should fail due to expected be equal than received 1`] = `
Object {
  "message": "expect([32mexpected[39m).toBeGreaterThan([31mreceived[39m)
result: 1 > 1.",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should fail due to expected be less than received 1`] = `
Object {
  "message": "expect([32mexpected[39m).toBeGreaterThan([31mreceived[39m)
result: 1 > 2.",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'expected' not to be a number 1`] = `
Object {
  "message": "value is not a number.
received: '[31mstring[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'expected' not to be a number 2`] = `
Object {
  "message": "value is not a number.
received: '[31mstring[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'expected' not to be a number 3`] = `
Object {
  "message": "value is not a number.
received: '[31mboolean[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'expected' not to be a number 4`] = `
Object {
  "message": "value is not a number.
received: '[31mboolean[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'expected' not to be a number 5`] = `
Object {
  "message": "value is not a number.
received: '[31marray[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'received' not to be a number 1`] = `
Object {
  "message": "received is not a number.
received: '[31mstring[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'received' not to be a number 2`] = `
Object {
  "message": "received is not a number.
received: '[31mstring[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'received' not to be a number 3`] = `
Object {
  "message": "received is not a number.
received: '[31mboolean[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'received' not to be a number 4`] = `
Object {
  "message": "received is not a number.
received: '[31mboolean[39m'",
  "pass": false,
}
`;

exports[`testing toBeGreaterThan should return false for 'received' not to be a number 5`] = `
Object {
  "message": "received is not a number.
received: '[31marray[39m'",
  "pass": false,
}
`;
